pkg/sql/conn_executor.go

```diff
@@ -2280,7 +2280,13 @@ func (ex *connExecutor) serialize() serverpb.Session {
 		if query.hidden {
 			continue
 		}
-		sql := truncateSQL(query.stmt.String())
+		sql, err := query.getStatement()
+		if err != nil {
+			// The statement recorded must have parsed correctly to get to
+			// this point. We should not error out accessing it right now.
+			panic(err)
```

Comment:
This is definitely a no-no - let's not panic here. We'd prefer to return an error, or serialize an error into the string, anything you want.

Reply:
sorry, I wasn't sure if it was possible to plumb an error down to this function, but it seems like it was possible.